Formula/python@2.rb

```diff
@@ -119,6 +127,17 @@ def install
       s.gsub! 'sqlite_defines.append(("SQLITE_OMIT_LOAD_EXTENSION", "1"))', ""
     end
 
+    # Fix setup.py so it handles absolute SDK paths properly.
+    # Current handling causes problems with zlib when using the CommandLineTools SDK.
+    # The problems does not yet manifest with Xcode SDK, because is_macosx_sdk_path()
+    # is not implemented correctly.
+    inreplace "setup.py" do |s|
+      s.gsub! "if host_platform == 'darwin' and is_macosx_sdk_path(dir):",
+              "if host_platform == 'darwin' and is_macosx_sdk_path(dir) and not dir.startswith(sysroot):"
+      s.gsub! "if host_platform == 'darwin' and is_macosx_sdk_path(zlib_h):",
+              "if host_platform == 'darwin' and is_macosx_sdk_path(zlib_h) and not zlib_h.startswith(macosx_sdk_root()):"
```

Comment:
This should be submitted in patch form, to match what is happening upstream, with a comment linking to the issue or PR upstream.

From your comment, it sounds like a fix is needed in the `is_macosx_sdk_path()` method rather than in `setup.py`. Is that true? Any idea how to patch that?